Dockerfile.aro-multistage-onebranch

```diff
@@ -4,7 +4,7 @@
 # OneBranch pipelines currently pull from CDPx which is deprecated. As a temporary fix before
 # we migrate to a new solution, this multistage dockerfile uses the same image in both steps to avoid
 # needing to pull images that aren't hosted in CDPx.
-FROM cdpxlinux.azurecr.io/user/aro/ubi8-gotoolset-1.17.7-13:20220526 AS builder
+FROM cdpxb8e9ef87cd634085ab141c637806568c00.azurecr.io/b8e9ef87-cd63-4085-ab14-1c637806568c/official/go-toolset-1.17.7-13:latest AS builder
```

Comment:
Should we be using `latest` here? It's generally a better idea to pin to a specific known-good version unless there's a good reason not to.

Reply:
We are pinned to a specific version here. Here's the Dockerfile that pulls this image from Red Hat and pushes it up to `cdpxb8e9ef87cd634085ab141c637806568c00.azurecr.io`: https://msazure.visualstudio.com/AzureRedHatOpenShift/_git/ARO.Build.Images?path=/ubi8-gotoolset-1.17.7-13/20220526/Dockerfile&version=GBmain&line=1&lineEnd=2&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

If it's confusing with to use the `:latest` tag, I can re-tag it with `:20220526` like we had before, but it's the same image.

Reply:
@cadenmarchese the onebranch.pipeline.imagebuildinfo task can take arguments to supply to the docker build command, see: https://dev.azure.com/onebranch/OneBranch/_wiki/wikis/OneBranch.wiki/4601/Build-Docker-Images.

You might want to consider that in a future change. That way you can pass in the params for registry and tag.

Reply:
Thanks for the additional context about the image version. Since it's pinned in the one you linked I don't think it's absolutely necessary here, but having it explicitly pinned makes it clear what's going on. Just a matter of preference at this point though.

Reply:
@dem4gus I'll plan on re-tagging the image in our isolated ACR, I think it would make the most sense to have `go-toolset:1.17.7`. What do you think?

@coreyperkins I was able to [successfully test](https://msazure.visualstudio.com/AzureRedHatOpenShift/_build/results?buildId=58827377&view=results) removing the onebranch dockerfile and using --build-args in the pipeline to supply the registry. This essentially brings us back to where we were before https://github.com/Azure/ARO-RP/pull/2126, but with the Isolated ACR referenced for the build container instead of the cdpxlinux ACR. Let me know if you think that would work for this change.

Reply:
> I'll plan on re-tagging the image in our isolated ACR, I think it would make the most sense to have `go-toolset:1.17.7`. What do you think?

That sounds like a good plan.

Reply:
@cadenmarchese lgtm!!

Reply:
cc @ross-bryan. The OneBranch Dockerfile was added in https://github.com/Azure/ARO-RP/pull/2126, because I was under the impression that every pull in every OneBranch pipeline needed to come from an ACR. This is no longer my assumption, but we do need the top-level build container to pull from an ACR.